cedar-policy/src/api.rs

```diff
@@ -2209,12 +2209,102 @@ impl FromStr for RestrictedExpression {
     }
 }
 
+/// Builder for a [`Request`]
+///
+/// Note that you can create the `EntityUid`s using `.parse()` on any
+/// string (via the `FromStr` implementation for `EntityUid`).
+/// The principal, action, and resource fields are optional to support
+/// the case where these fields do not contribute to authorization
+/// decisions (e.g., because they are not used in your policies).
+/// If any of the fields are `None`, we will automatically generate
+/// a unique entity UID that is not equal to any UID in the store.
+/// 
+/// The default for principal, action and resource fields is Unknown.
+#[derive(Debug, Default)]
+pub struct RequestBuilder {
+    principal: Option<ast::EntityUIDEntry>,
+    action: Option<ast::EntityUIDEntry>,
+    resource: Option<ast::EntityUIDEntry>,
+    context: Option<ast::Context>,
+}
+
+impl RequestBuilder {
+    /// Set the principal
+    pub fn principal(self, principal: Option<EntityUid>) -> Self {
+        Self {
+            principal: Some(match principal {
+                Some(p) => ast::EntityUIDEntry::concrete(p.0),
+                None => ast::EntityUIDEntry::concrete(ast::EntityUID::unspecified_from_eid(
```

Comment:
Looks like this PR is using `unspecified` entities, not unknown entities. These are slightly different concepts. Which ones are you trying to use? Note that `unknown` entities will require moving this change behind the partial evaluation feature falg

Reply:
`unspecified` was kept for backwards compatibility with the current `Request::new` api. The specific change is
in the `RequestBuilder::build` function which will use `unknown` in case no `principal`, `action` or `resource` has been specified at all.

I am evaluating `cedar` in a service which has some specific pagination (including grouping and reporting count based on group keys) calls. The goal is to use `unknown` for the `resource` in combination with the partial api and to translate the residue policies to mongodb queries.

I could put the whole `RequestBuilder` behind the feature gate, what do you think?

Reply:
Ah I see, sorry for my confusion then! 
I think putting the whole thing behind that feature is the easiest option, unless you want to try to break it out into some functions outside the feature some functions inside the feature. 
In general though, love the change and excited to see people experimenting with pe. Would love feedback as you continue to work on it.

Reply:
Yep, feel free to ask questions & provide feedback on slack. You can also file issues in this repo if you have specific suggestions for improvement :)